code-challenges.test.js

```diff
@@ -15,42 +17,179 @@
-// Expected output: [-823, 7, 23]
+        
 
+    expect(lowHiNeg(fullArr1, "number")).toEqual([-9, 7, 9, 199])
+    expect(lowHiNeg(fullArr2, "number")).toEqual([-823, 7, 23])
+    
+})
+})
+
+
+// FAIL  ./code-challenges.test.js
+// fibArr
+//   ✓ takes a number greater than 2 and returns Fibonacci sequence in an array (1 ms)
+// lowHiNeg
+//   ✕ takes in an array and returns an new array of only odd numbers sorted from least to greatest
+
+// ● lowHiNeg › takes in an array and returns an new array of only odd numbers sorted from least to greatest
 
 // b) Create the function that makes the test pass.
 
 
+//-------------------------------------- PSEUDO CODE
+// create a function called hiloNeg that takes in array
+// use .filter to filter dataType of numbers then filter the odd numbers 
+// use sort to sort new array from lowest to highest 
+
+
+
+const lowHiNeg = (array, dataType) => {
+
+    let final = array.filter((value) => {
+        return typeof value === dataType})
+            let finalArr = final.filter((value) => value % 2 != 0)
+                .sort((a,b) => a-b)
+                return finalArr
+}
+
+
 // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const numbersToAdd1 = [2, 4, 45, 9]
-// Excpected output: [2, 6, 51, 60]
 
-const numbersToAdd2 = [0, 7, -8, 12]
-// Expected output: [0, 7, -1, 11]
 
-const numbersToAdd3 = []
-// Expected output: []
+describe("cumulativeSum", () => {
+    it("takes in an array of accumulating sum", () => {
+        const numbersToAdd1 = [2, 4, 45, 9]
+        const numbersToAdd2 = [0, 7, -8, 12]
+        const numbersToAdd3 = []
+
+    expect(cumulativeSum(numbersToAdd1)).toEqual([2, 6, 51, 60])
+    expect(cumulativeSum(numbersToAdd2)).toEqual([0, 7, -1, 11])
+    expect(cumulativeSum(numbersToAdd3)).toEqual([])
+
+    
+})
+})
+
+
+
+// PASS  ./code-challenges.test.js
+// fibArr
+//   ✓ takes a number greater than 2 and returns Fibonacci sequence in an array (1 ms)
+// lowHiNeg
+//   ✓ takes in an array and returns an new array of only odd numbers sorted from least to greatest
+// cumulativeSum
+//   ✓ takes in an array of accumulating sum (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       3 passed, 3 total
+
+
+
 
 
 // b) Create the function that makes the test pass.
+
+//------------------------- Pseudo 
+
+// create function cumulativeSum
+// use arr paramater
+// resourced cumalative sum that can return an empty array. !== [undefined]
+
+
+const numbersToAdd1 = [2, 4, 45, 9];
+const numbersToAdd2 = [0, 7, -8, 12];
+const numbersToAdd3 = [];
+
+const cumulativeSum = (arr) =>
+    arr.map(
+        (
+            (sum) => (value) =>
+                (sum += value)
+        )(0)
+    );
+//console.log(accumulate(numbersToAdd3))
```

Comment:
This is good BUT your variables should only exist in your test and not floating out by themselves.